interfaces/console/src/app/console/sites/[id]/page.tsx

```diff
@@ -580,104 +379,171 @@ const Page: React.FC<SiteDetailsProps> = ({ params }) => {
+        if (
+          topic.startsWith(`stat-${user.orgName}-${user.id}-${Stats_Type.Site}`)
+        ) {
+          PubSub.unsubscribe(topic);
+          delete subscriptionsRef.current[topic];
+        }
+      });
+
+      subscriptionsRef.current[sKey] = true;
+
       getSiteMetricStat({
         variables: {
           data: {
-            to: to,
-            siteId: id,
-            from: from,
+            to,
+            from,
             userId: user.id,
             step: STAT_STEP_29,
             orgName: user.orgName,
             withSubscription: true,
             type: Stats_Type.Site,
+            siteIds: [id],
+            nodeIds: nodeIds,
+          },
+        },
+      });
+
+      MetricStatBySiteSubscription({
+        key: sKey,
+        siteIds: [id],
+        userId: user.id,
+        url: env.METRIC_URL,
+        orgName: user.orgName,
+        type: Stats_Type.Site,
+        from,
+        nodeIds: nodeIds,
+      });
+
+      return () => {
+        PubSub.unsubscribe(sKey);
+        delete subscriptionsRef.current[sKey];
+      };
+    }
+  }, [
+    id,
+    nodeIds,
+    nodesFetched,
+    user.id,
+    user.orgName,
+    env.METRIC_URL,
+    getSiteMetricStat,
+  ]);
+
+  useEffect(() => {
+    if (metricFrom > 0 && id) {
+      getMetricBySite({
+        variables: {
+          data: {
+            step: 30,
+            siteId: id,
+            userId: user.id,
+            type: graphType,
+            from: metricFrom,
+            orgName: user.orgName,
+            withSubscription: true,
+            to: metricFrom + METRIC_RANGE_10800,
           },
         },
       });
     }
+  }, [metricFrom, getMetricBySite, graphType, id, user.id, user.orgName]);
+
+  const handleSiteStatSubscription = useCallback(
+    (_: any, data: string) => {
+      try {
+        const parsedData: TMetricResDto = JSON.parse(data);
+        if (parsedData?.data?.getSiteMetricStatSub) {
+          const metric = parsedData.data.getSiteMetricStatSub;
+          const { type, success, siteId, nodeId, value } = metric;
+
+          if (success && siteId === id) {
+            if (!nodeId) {
+              if (type === SITE_KPI_TYPES.SITE_UPTIME) {
+                PubSub.publish(
+                  `${TOPIC_PREFIXES.SITE_UPTIME_STAT}-${siteId}`,
+                  Math.floor(value[1]),
+                );
+              } else if (type === SITE_KPI_TYPES.SITE_UPTIME_PERCENTAGE) {
+                PubSub.publish(
+                  `${TOPIC_PREFIXES.SITE_UPTIME_PERCENTAGE_STAT}-${siteId}`,
+                  Math.floor(value[1]),
+                );
+              }
+            }
+            PubSub.publish(`stat-${type}`, value);
+
+            if (nodeId && type === SITE_KPI_TYPES.NODE_UPTIME) {
+              PubSub.publish(
+                `${TOPIC_PREFIXES.NODE_UPTIME_STAT}-${nodeId}-${siteId}`,
+                Math.floor(value[1] || 1),
+              );
+            }
+          }
```

Comment:
This can be improved if keep topic name simple, I hope we are using sam metric keys on console as we have defined on backend. If that's the case the topic can be as simple as
`PubSub.publish(`stat-${type}-${siteId}-${nodeId}`, value);`
We dont need to check id its a `SITE_UPTIME_PERCENTAGE_STAT` or `SITE_UPTIME_PERCENTAGE `

Reply:
Addressed.